### PR TITLE
Change build process to demo Azure DevOps integration

### PR DESCRIPTION
**Pull request by garoyeri**
Demonstrate how we would setup a build script for a bigger project and integrate with Azure DevOps for building and testing.

Use Psake to create tasks that can be used during development and partition the build script
Use azure-pipelines.yml to define Azure DevOps build pipeline
Update README with more details on how to operate the build script
Use GitVersion to generate unique version numbers for each build and apply them as Informational Versions for the assemblies

### CODE REVIEW

File: ContosoUniversity.sln [hunk 1]
```diff
@@ -9,11 +9,18 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "ContosoUniversity.Integrati
 EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Solution Items", "Solution Items", "{13186C93-7307-4FEA-A17C-4556F54FF5F1}"
 	ProjectSection(SolutionItems) = preProject
-		appveyor.yml = appveyor.yml
-		Build.ps1 = Build.ps1
-		Init.ps1 = Init.ps1
+		.gitattributes = .gitattributes
+		.gitignore = .gitignore
+		azure-pipelines.yml = azure-pipelines.yml
```
Comment (jbogard) [inline ContosoUniversity.sln]: Make these GitHub actions and I'll merge :)

Comment (garoyeri) [reply ContosoUniversity.sln]: Sure thing. Can we keep the azure-pipelines.yml as an example?
Do you still want to support appveyor?

Comment (jbogard) [reply ContosoUniversity.sln]: So I thiiiiiink if we wanted an AzDO example, I'd create an AzDO project with pipelines etc. The goal here wasn't really to highlight CI/CD - I have a separate repo for that on dev.azure.com already. Mainly it was for validation of PRs here.

Comment (garoyeri) [reply ContosoUniversity.sln]: Alright, I went ahead and pushed an update with GitHub Actions. It doesn't have all the same features, but at least provides the whole experience and a pass / fail with some artifacts. There are some nice extensions to this that are missing (Azure Pipelines has more out of the box) but I think it'll get you moving at least.

File: setup.sh [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env pwsh
```
Comment (jbogard) [inline setup.sh]: Does this actually work on a mac? Since I use SQL Local DB that is

Comment (jbogard) [reply setup.sh]: Narrator: it does not

```
RoundhousE encountered an error.
System.PlatformNotSupportedException: LocalDB is not supported on this platform.
```

Comment (garoyeri) [reply setup.sh]: Yeah, no, it doesn't. But if you use PostgreSQL ... *hack* *cough* *sputter*

**Pull request merged**